src/envoy/http/metadata_exchange/config.h

```diff
@@ -0,0 +1,83 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "extensions/common/wasm/null/null.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Wasm {
+namespace MetadataExchange {
+
+constexpr char ExchangeMetadataHeader[] = "x-envoy-peer-metadata";
+
+// NodeMetadata key is the key in the node metadata struct that is passed
+// between peers.
+constexpr char NodeMetadataKey[] = "istio.io/metadata";
+
+// DownstreamMetadataKey is the key in the request metadata for downstream peer
+// metadata
+constexpr char DownstreamMetadataKey[] =
+    "envoy.wasm.metadata_exchange.downstream";
+
+// UpstreamMetadataKey is the key in the request metadata for downstream peer
+// metadata
+constexpr char UpstreamMetadataKey[] = "envoy.wasm.metadata_exchange.upstream";
+
+using StringView = absl::string_view;
```

Comment:
In header file, generally it's not good to introduce classes to namespace by keyword `using` . Not sure if there is exception.

Reply:
Macro expansion below requires it... I agree this is not a good style, but this is a problem in the upstream.